argostime/crawler/ekoplaza.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+"""
+    crawler/ekoplaza.py
+
+    Crawler for ekoplaza.nl
+
+    Copyright (c) 2022 Anna <anna [at] anna.computer>
+
+    This file is part of Argostimè.
+
+    Argostimè is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Argostimè is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Argostimè. If not, see <https://www.gnu.org/licenses/>.
+"""
+
+
+import json
+import logging
+
+
+import requests
+
+from argostime.exceptions import CrawlerException
+from argostime.exceptions import PageNotFoundException
+from argostime.crawler.crawl_utils import CrawlResult
+
+
+def crawl_ekoplaza(url: str):  # -> CrawlResult:
+    """Ekoplaza crawler"""
+
+    info = url.split('product/')[-1]
+    response = requests.get(
+        f'https://www.ekoplaza.nl/api/aspos/products/url/{info}')
+
+    if response.status_code != 200:
+        logging.error("Got status code %d while getting url %s",
+                      response.status_code, url)
+        raise PageNotFoundException(url)
+    try:
+        product = response.json()["Product"]
+    except KeyError as exception:
+        logging.error("No product found at %s", url)
+        raise CrawlerException from exception
+
+    result = CrawlResult(url=url)
+
+    result.url = url
+    try:
+        result.product_name = product['Description'].title()
+    except KeyError as exception:
+        logging.error("No product name found in %s", product)
+        raise CrawlerException from exception
+
+    try:
+        result.product_code = product['DefaultScanCode']['Code']  # EAN ?
```

Comment:
I checked a few products and these seem to be valid EAN codes for the products.

Reply:
So we should use EAN for both then right? So we can use that EAN in the future to match the same products across multiple shops.

Reply:
Yes exactly